### ISSUE TITLE
Terraform fails when using rancher2_cluster with cluster_monitoring_input

### ISSUE THREAD

**Issue by TheBigLee** [rancher/terraform-provider-rancher2#421]
Hi

I tried to import an existing GKE cluster into rancher using the `rancher2_cluster` resource.
The terraform code looks like this:
```
provider "rancher2" {
  api_url    = var.RANCHER_API_URL
  access_key = var.RANCHER_ACCESS_KEY
  secret_key = var.RANCHER_SECRET_KEY
}

resource "rancher2_cluster" "gke" {
  name                      = var.cluster_name
  enable_cluster_monitoring = "true"
  cluster_monitoring_input {
    answers = var.cluster_monitoring_answers
  }
}
```
The cluster_monitoring_answers variable looks like this:
```
cluster_monitoring_answers = {
    "exporter-kubelets.https" = true

    "exporter-node.enabled"                 = true
    "exporter-node.ports.metrics.port"      = 9796
    "exporter-node.resources.limits.cpu"    = "400m"
    "exporter-node.resources.limits.memory" = "50Mi"

    "grafana.persistence.enabled"      = true
    "grafana.persistence.size"         = "10Gi"
    "grafana.persistence.storageClass" = "standard"

    "operator-init.enabled" = true

    "operator.resources.limits.memory" = "100Mi"

    "prometheus.persistence.enabled"            = true
    "prometheus.persistence.size"               = "50Gi"
    "prometheus.persistence.storageClass"       = "standard"
    "prometheus.resources.core.limits.cpu"      = "1000m"
    "prometheus.resources.core.limits.memory"   = "1500Mi"
    "prometheus.resources.core.requests.cpu"    = "750m"
    "prometheus.resources.core.requests.memory" = "750Mi"
    "prometheus.retention"                      = "336h"
}
```
This leads to the following error when running `terraform apply`
```
Error: action [editMonitoring] not available on [&{h-i4uot cluster map[apiServices:https://rancher.mydomain.tld/v3/clusters/h-i4uot/apiservices clusterAlertGroups:https://rancher.mydomain.tld/v3/clusters/h-i4uot/clusteralertgroups clusterAlertRules:https://rancher.mydomain.tld/v3/clusters/h-i4uot/clusteralertrules clusterAlerts:https://rancher.mydomain.tld/v3/clusters/h-i4uot/clusteralerts clusterCatalogs:https://rancher.mydomain.tld/v3/clusters/h-i4uot/clustercatalogs clusterLoggings:https://rancher.mydomain.tld/v3/clusters/h-i4uot/clusterloggings clusterMonitorGraphs:https://rancher.mydomain.tld/v3/clusters/h-i4uot/clustermonitorgraphs clusterRegistrationTokens:https://rancher.mydomain.tld/v3/clusters/h-i4uot/clusterregistrationtokens clusterRoleTemplateBindings:https://rancher.mydomain.tld/v3/clusters/h-i4uot/clusterroletemplatebindings clusterScans:https://rancher.mydomain.tld/v3/clusters/h-i4uot/clusterscans etcdBackups:https://rancher.mydomain.tld/v3/clusters/h-i4uot/etcdbackups namespaces:https://rancher.mydomain.tld/v3/clusters/h-i4uot/namespaces nodePools:https://rancher.mydomain.tld/v3/clusters/h-i4uot/nodepools nodes:https://rancher.mydomain.tld/v3/clusters/h-i4uot/nodes notifiers:https://rancher.mydomain.tld/v3/clusters/h-i4uot/notifiers persistentVolumes:https://rancher.mydomain.tld/v3/clusters/h-i4uot/persistentvolumes projects:https://rancher.mydomain.tld/v3/clusters/h-i4uot/projects remove:https://rancher.mydomain.tld/v3/clusters/h-i4uot self:https://rancher.mydomain.tld/v3/clusters/h-i4uot shell:wss://rancher.mydomain.tld/v3/clusters/h-i4uot?shell=true storageClasses:https://rancher.mydomain.tld/v3/clusters/h-i4uot/storageclasses subscribe:https://rancher.mydomain.tld/v3/clusters/h-i4uot/subscribe templates:https://rancher.mydomain.tld/v3/clusters/h-i4uot/templates tokens:https://rancher.mydomain.tld/v3/clusters/h-i4uot/tokens update:https://rancher.mydomain.tld/v3/clusters/h-i4uot] map[generateKubeconfig:https://rancher.mydomain.tld/v3/clusters/h-i4uot?action=generateKubeconfig importYaml:https://rancher.mydomain.tld/v3/clusters/h-i4uot?action=importYaml viewMonitoring:https://rancher.mydomain.tld/v3/clusters/h-i4uot?action=viewMonitoring]}]
  on modules/cluster/rancher.tf line 1, in resource "rancher2_cluster" "gke":
   1: resource "rancher2_cluster" "gke" {
```

If I uncomment the lines
```
cluster_monitoring_input {
    answers = var.cluster_monitoring_answers
  }
```
it works perfectly and if I remove the comments afterwards and rerun `terraform apply`  the correct monitoring configurations are safed in rancher.

I therefore suspect some kind of race condition.

**Comment by rawmind0** [comment]
Hi @TheBigLee , this issue has been addressed with PR #409 . It's already merged on master and it will be included on next provider release.

**Comment by TheBigLee** [comment]
Hi @rawmind0 
Thanks a lot. I didn't realize that this was already solved.
I will close this issue.
